### DISCUSSION TITLE
Moon's position angle?

### DISCUSSION THREAD

**Discussion by CoolTowel** [cosinekitty/astronomy#278] [Q&A] [answered]
I want to calculate the position angle like that in this [NASA page](https://svs.gsfc.nasa.gov/5048). This position angle (I think) is the position angle of north pole with respect to the moon's location.

So I used the `Astronomy_RotationAxis` function to get the vector that pointing in the direction of the body's north pole in J2000 as described [here](https://github.com/cosinekitty/astronomy/tree/master/source/c#astro_axis_t). Then I scaled it to a reasonable length (the radius of moon in AU), and added it to the vector of moon's location in geocentric Cartesian coordinates (given by `Astronomy_GeoVector`). Then I think I got the vector moon's north pole location. Finally I converted both vector to ra dec, and used the `iauPas` function provided by [SOFA](https://www.iausofa.org/) to calculate the position angle. 

But I can't get the correct value, I got the value varies between about +-1.5 degrees. But the correct value should varies between +-20 degrees (roughly the the angle between equator and moon's orbit plane)

I wrote above code in Swift (which can easily use C libraries). 

    var oneAstronomyTime = Astronomy_MakeTime(2023, 1, 1, 00, 00, 00.0)

    let moonAxis = withUnsafeMutablePointer(to: &oneAstronomyTime) {timeptr in
        Astronomy_RotationAxis(BODY_MOON, timeptr) // moon axis info
    }
    
    let moonVector = Astronomy_GeoVector(BODY_MOON, oneAstronomyTime, NO_ABERRATION) // moon's location vector
    
    var moonNorthVector: astro_vector_t! = astro_vector_t() // moon north pole location vector in geocentric coords
    moonNorthVector.status = ASTRO_SUCCESS
    
    var scale: Double = 0.0000116138 // moon radius in au
    moonNorthVector.x = moonVector.x + moonAxis.north.x * scale
    moonNorthVector.y = moonVector.y + moonAxis.north.y * scale
    moonNorthVector.z = moonVector.z + moonAxis.north.z * scale // moon location vector add the north pole vector
    moonNorthVector.t = moonVector.t // set the same time
    
    var moonEQ = Astronomy_EquatorFromVector(moonVector) // moon location in ra dec
    var moonNorthEQ = Astronomy_EquatorFromVector(moonNorthVector) // moon north pole location in ra dec

    var psa = iauPas(degToRad(moonEQ.ra), degToRad(moonEQ.dec), degToRad(moonNorthEQ.ra), degToRad(moonNorthEQ.dec)) // iau standard function calculate the position angle in rad
    
    return radToDeg(psa) // give the position angle in deg

**Comment by CoolTowel** [comment]
I just found that the value I got are smaller by a factor of 14.5 compared with the NASA value, that is weird...

<img width="711" alt="image" src="https://user-images.githubusercontent.com/37588427/212085845-79ab3730-6dcf-4baa-882d-7e144c6d79a9.png">

<img width="1204" alt="image" src="https://user-images.githubusercontent.com/37588427/212085941-dbaecc5e-a3b6-4990-8c07-be84f212ee6b.png">

**Comment by cosinekitty** [comment, marked as answer]
One thing that will affect your calculations is that right ascension (RA) is not measured in degrees, but sidereal hours. To convert RA to degrees, you need to multiply it by (360/24) = 15. Can you try making that correction and see how it affects your results?

**Reply by CoolTowel** [reply]
Yes that‘s the reason. Thank you very much! I didn't check the `astro_equatorial_t` structure carefully

**Reply by cosinekitty** [reply]
I'm glad that helped!
